class io.netty.resolver.RoundRobinInetAddressResolver$1 extends java.lang.Object implements io.netty.util.concurrent.FutureListener
{
final io.netty.util.concurrent.Promise val$promise;
final java.lang.String val$inetHost;
final io.netty.resolver.RoundRobinInetAddressResolver this$0;
void <init>(io.netty.resolver.RoundRobinInetAddressResolver, io.netty.util.concurrent.Promise, java.lang.String)
{
io.netty.util.concurrent.Promise v2;
io.netty.resolver.RoundRobinInetAddressResolver$1 v0;
java.lang.String v3;
io.netty.resolver.RoundRobinInetAddressResolver v1;
v0 := @this: io.netty.resolver.RoundRobinInetAddressResolver$1;
v1 := @parameter0: io.netty.resolver.RoundRobinInetAddressResolver;
v2 := @parameter1: io.netty.util.concurrent.Promise;
v3 := @parameter2: java.lang.String;
v0.<io.netty.resolver.RoundRobinInetAddressResolver$1: io.netty.resolver.RoundRobinInetAddressResolver this$0> = v1;
v0.<io.netty.resolver.RoundRobinInetAddressResolver$1: io.netty.util.concurrent.Promise val$promise> = v2;
v0.<io.netty.resolver.RoundRobinInetAddressResolver$1: java.lang.String val$inetHost> = v3;
specialinvoke v0.<java.lang.Object: void <init>()>();
return;
}
public void operationComplete(io.netty.util.concurrent.Future) throws java.lang.Exception
{
java.lang.Throwable v12;
io.netty.util.concurrent.Promise v11, v5, v8;
io.netty.resolver.RoundRobinInetAddressResolver$1 v0;
java.net.UnknownHostException v9;
int v4, v6;
java.lang.Object v3, v7;
java.lang.String v10;
io.netty.util.concurrent.Future v1;
boolean v2;
v0 := @this: io.netty.resolver.RoundRobinInetAddressResolver$1;
v1 := @parameter0: io.netty.util.concurrent.Future;
v2 = interfaceinvoke v1.<io.netty.util.concurrent.Future: boolean isSuccess()>();
if v2 == 0 goto label2;
v3 = interfaceinvoke v1.<io.netty.util.concurrent.Future: java.lang.Object getNow()>();
v4 = interfaceinvoke v3.<java.util.List: int size()>();
if v4 <= 0 goto label1;
v5 = v0.<io.netty.resolver.RoundRobinInetAddressResolver$1: io.netty.util.concurrent.Promise val$promise>;
v6 = staticinvoke <io.netty.resolver.RoundRobinInetAddressResolver: int randomIndex(int)>(v4);
v7 = interfaceinvoke v3.<java.util.List: java.lang.Object get(int)>(v6);
interfaceinvoke v5.<io.netty.util.concurrent.Promise: io.netty.util.concurrent.Promise setSuccess(java.lang.Object)>(v7);
goto label3;
label1:
v8 = v0.<io.netty.resolver.RoundRobinInetAddressResolver$1: io.netty.util.concurrent.Promise val$promise>;
v9 = new java.net.UnknownHostException;
v10 = v0.<io.netty.resolver.RoundRobinInetAddressResolver$1: java.lang.String val$inetHost>;
specialinvoke v9.<java.net.UnknownHostException: void <init>(java.lang.String)>(v10);
interfaceinvoke v8.<io.netty.util.concurrent.Promise: io.netty.util.concurrent.Promise setFailure(java.lang.Throwable)>(v9);
goto label3;
label2:
v11 = v0.<io.netty.resolver.RoundRobinInetAddressResolver$1: io.netty.util.concurrent.Promise val$promise>;
v12 = interfaceinvoke v1.<io.netty.util.concurrent.Future: java.lang.Throwable cause()>();
interfaceinvoke v11.<io.netty.util.concurrent.Promise: io.netty.util.concurrent.Promise setFailure(java.lang.Throwable)>(v12);
label3:
return;
}
}